server/messages.go

```diff
@@ -141,20 +143,27 @@ func processMessages(conn net.Conn, c chan *packetInfo, t *Tracker) {
 		}
 
 		if err := scanner.Err(); err != nil {
-			fmt.Fprintf(os.Stderr, "[Error] %s\n", err.Error())
-			break
+			fmt.Fprintf(os.Stderr, logCommunication+"Error scanning message: %s\n", err)
+			return
 		}
 
 		if mb.Len() == 0 {
-			break
+			fmt.Fprint(os.Stderr, logCommunication+"0-length message\n")
+			return
 		}
 
 		// Extend the deadline, we got a valid full message.
-		conn.SetDeadline(time.Now().Add(deadline))
+		if err := conn.SetDeadline(time.Now().Add(deadline)); err != nil {
+			// Failing to set the deadline could be due to the client getting
+			// disconnected for some reason. Do not consider the read itself
+			// a failure due to failure to set the deadline. The client will drop
+			// off eventually after connection is broken anyway.
+			fmt.Printf(logCommunication+"Error setting deadline after successful receive: %s\n", err)
+		}
```

Comment:
This is the same as the send+disconnect race condition. Even though the setdeadline might fail with a disconnect, we still want to handle the valid received message.